testdata/p4_16_samples_outputs/strength4-first.p4

```diff
@@ -0,0 +1,55 @@
+#include <core.p4>
+#include <v1model.p4>
+
+header ethernet_t {
+    bit<48> dst_addr;
+    bit<48> src_addr;
+    bit<16> ether_type;
+}
+
+struct my_headers_t {
+    ethernet_t ethernet;
+}
+
+typedef my_headers_t headers_t;
+struct local_metadata_t {
+    bit<16> f16;
+    bit<16> m16;
+    bit<16> d16;
+}
+
+parser parser_impl(packet_in packet, out headers_t hdr, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
+    state start {
+        transition accept;
+    }
+}
+
+control ingress_impl(inout headers_t hdr, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
+    apply {
+    }
+}
+
+control egress_impl(inout headers_t hdr, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
+    apply {
+        local_metadata.m16 = (16w0 ++ local_metadata.f16)[14:0] ++ 1w0;
```

Comment:
this seems sub-optimal, you should be able to drop the concat altogether

Reply:
it is simplified in testdata/p4_16_samples_outputs/strength4-midend.p4

Reply:
I guess because the midend runs StrengthReduction again?  So if we had a recursive call here, it would get simplified in the frontend...